File: app/src/main/java/com/oneflow/analytics/model/adduser/OFAddUserResponse.java
/*
 *  Copyright 2021 1Flow, Inc.
 *
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.oneflow.analytics.model.adduser;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.oneflow.analytics.model.OFBaseResponse;

public class OFAddUserResponse extends OFBaseResponse {

    @SerializedName("result")
    @Expose
    private OFAddUserResultResponse result;

    public OFAddUserResultResponse getResult() {
        return result;
    }

    public void setResult(OFAddUserResultResponse result) {
        this.result = result;
    }
}

File: app/src/main/java/com/oneflow/analytics/model/adduser/OFAddUserResponse.kt
package com.oneflow.analytics.model.adduser

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "User")
data class OFAddUserResponse(
        @PrimaryKey
        @ColumnInfo(name = "analytic_user_id")
        @SerializedName("analytic_user_id")
        var analytic_user_id: String,

        @ColumnInfo(name = "first_name")
        @SerializedName("first_name")
        val first_name: String?,

        @ColumnInfo(name = "last_name")
        @SerializedName("last_name")
        val last_name: String? = null,

        @ColumnInfo(name = "system_id")
        @SerializedName("system_id")
        var system_id: String? = null,

        @ColumnInfo(name = "created_on")
        @SerializedName("created_on")
        var created_on: Long,

        @ColumnInfo(name = "updated_on")
        @SerializedName("updated_on")
        var updated_on: Long
)
